integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala

```diff
@@ -208,7 +209,10 @@ class CarbonFileMetastore extends CarbonMetaStore {
     try {
       lookupRelation(tableIdentifier)(sparkSession)
     } catch {
-      case _: Exception =>
+      case ex: Exception =>
+        if (ex.getCause.isInstanceOf[HiveException]) {
+          throw ex
+        }
```

Comment:
i suggest if you are  skipping the exception propagation, better you log the reason why you are returning false.
this will be better for a secondary developer to locate the exact problem from a log  in some unwanted failure scenarios.

Reply:
I think it will better to have a testcases here? i can see from your code in CarbonFileMetastore.scala  that you are only expecting a HiveException type to be thrown out, other exceptions you are ignoring.

Reply:
i checked the flow and permission errors will be thrown by Hive as HiveException and wrapped as analysis exception, and all the other exceptions will be same as i have described in description, during that time we cannot just catch exception and return false as table not exists. so i m checking as HiveException

Reply:
Didn't got your context of explanation is  for which review comment?

Reply:
here we arereturning false directly after catching exception, telling the table does not exists, this is wrong, i have check the flow , we might get hiveException(regarding permission) also if the user is not allowed to access the table, that time also we return false and we will not get the proper error.

Reply:
There are exceptions from different hierarchy incase of table not found, not all are HiveException.
Ex:- NoSuchTableException extends  AnalysisException, InvalidTableException extends HiveException.
So, I think we can refer the exceptions from Spark Code related to table not exists, not found and return false for only those exception cases. All other exceptions we can throw back to caller.